core/appmodule/module.go

```diff
@@ -4,16 +4,13 @@ import (
 	"context"
 
 	"google.golang.org/grpc"
-
-	"cosmossdk.io/depinject"
 )
 
 // AppModule is a tag interface for app module implementations to use as a basis
 // for extension interfaces. It provides no functionality itself, but is the
 // type that all valid app modules should provide so that they can be identified
 // by other modules (usually via depinject) as app modules.
 type AppModule interface {
-	depinject.OnePerModuleType
```

Comment:
We need to update the docs for app wiring because of this. This will add one step to the integration of a module with depinject.

Reply:
Imho this makes the depinject UX a tiny bit worse, but I guess good documentation will solve that.

Can we maybe add a comment on AppModule as well mentioning this?

Reply:
yea can add the docs, its worse ux but cleans up things for us